### PR TITLE
Updated PR for switching to src layout

### PR DESCRIPTION
**Pull request by bsweger**
An alternate attempt to refactor the code base into src layout while not breaking where Pulumi thinks things should be.

Best not to review this one commit by commit: there's some thrashing, as I realized that PDM + Pulumi isn't an ideal combination for GitHub workflows (they seem to have specific, conflicting ideas about how to handle python virtual environments).

Apologies for the size of the PR. The actual code used for the Pulumi demo hasn't really changed. Most of the diffs are related to:

* moving Pulumi's "hello world" tutorial code into a src-based layout and adding a `pyproject.toml` file
* using [pip-tools](https://hynek.me/til/pip-tools-and-pyproject-toml/) to generate requirements files from the dependencies specified in `pyproject.toml`
* adding a workflow for code checks (linting and type checking, with tests to come soon)
* formatting changes to appease the linter
* README updates

### CODE REVIEW

File: Pulumi.yaml [hunk 1]
```diff
@@ -1,9 +1,8 @@
 name: hubverse-aws
 runtime:
   name: python
-  options:
-    virtualenv: venv
 description: Hubverse infrastructure for AWS
+main: src/hubverse_infrastructure/__main__.py
```
Comment (bsweger) [inline Pulumi.yaml:5]: Tell Pulumi to look for its project in the new `src` layout

File: README.md
Comment (bsweger) [inline README.md:1]: The diff on this file is confusing. Essentially, it's the same README as the prior PR (#30), with the addition of some notes for adding project dependencies, now that we're not using PDM.

File: environment.yml
Comment (bsweger) [inline environment.yml:1]: Now that we tossed PDM, replaced that setup process with a conda-based one that installs the correct version of Python and installs project dependencies from the project's requirements files.

File: requirements/dev-requirements.txt
Comment (bsweger) [inline requirements/dev-requirements.txt:1]: `pip-tools` generated this file by using the dependency list from `pyproject.toml`

(`python -m piptools compile -o requirements/requirements.txt pyproject.toml`)

File: src/hubverse_infrastructure/hubs/iam.py
Comment (bsweger) [inline src/hubverse_infrastructure/hubs/iam.py:1]: Only changes in this file are to comply with the linter

File: src/hubverse_infrastructure/hubs/s3.py
Comment (bsweger) [inline src/hubverse_infrastructure/hubs/s3.py:1]: Only changes in this file are to comply with the linter

**Comment by bsweger** [comment]
@elray1 This is an updated version of the PR you looked at earlier. They may indeed be a way to make Pulumi and PDM work together in a GitHub action, but I didn't want to spend any more time on it.

File: .gitignore
Comment (bsweger) [inline .gitignore:1]: Grabbed a more inclusive Python-based .gitignore from GitHub

File: pdm.lock
Comment (elray1) [inline pdm.lock:1]: Should we delete this `pdm.lock` file?

**Review by elray1 — COMMENTED** [review]
- Successfully setup my dev environment
- One question about whether we should remove the pdm.lock file
- Read through all other changes and they seemed reasonable

File: pdm.lock
Comment (bsweger) [reply pdm.lock:1]: Good catch--I missed this when messing around and just pushed an update.

**Review by elray1 — APPROVED** [review]

**Comment by elray1** [comment]
approved, will let you merge or squash/merge as you see fit!

**Pull request merged**